File: ios/MDCApp/Flanker/FlankerViewManager.m
//
//  TestViewManager.m
//  MDCApp
//
//  Created by Volodymyr Nazarkevych on 21.05.2022.
//  Copyright © 2022 Facebook. All rights reserved.
//

#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(RCTFlankerViewManager, RCTViewManager)
RCT_EXTERN_METHOD(setGameParameters:(NSString *)json)
RCT_EXTERN_METHOD(preloadGameImages:(NSString *)json)
RCT_EXTERN_METHOD(startGame:(BOOL *)isFirst isLast:(BOOL *)isLast)
RCT_EXPORT_VIEW_PROPERTY(onEndGame, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(dataJson, NSString)
RCT_EXPORT_VIEW_PROPERTY(onUpdate, RCTDirectEventBlock)
@end

File: ios/MDCApp/Flanker/FlankerViewManager.swift
import Foundation

@objc(RCTFlankerViewManager)
class FlankerViewManager: RCTViewManager {
  private let resultManeger = ResultManager.shared
  private var flankerView: FlankerView? = nil
    
  override func view() -> UIView! {
    flankerView = FlankerView()
    return flankerView
  }

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc
  func setGameParameters(_ json: String) {
    ParameterGameManager.shared.setJsonWithParameters(json: json)
  }

  @objc
  func preloadGameImages(_ json: String) {
    ParameterGameManager.shared.loadAllImage(dataJson: json)
  }

  @objc
  func startGame(_ isFirst: Bool, isLast: Bool) {
    if isFirst {
      flankerView!.typeResult = .ok
      flankerView!.isLast = false // todo - review if we need to use this
    } else if isLast {
      flankerView!.typeResult = .finish
    } else {
      flankerView!.typeResult = .next
    }
    flankerView!.parameterGame()
  }
}